kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -259,8 +263,16 @@ private static GenericKubernetesResource filterResourceFromRef(VirtualKafkaClust
 
     private static VirtualCluster getVirtualCluster(KafkaProxy primary,
                                                     VirtualKafkaCluster cluster,
-                                                    int clusterNum) {
-        String bootstrap = cluster.getSpec().getTargetCluster().getBootstrapping().getBootstrapAddress();
+                                                    int clusterNum, List<KafkaClusterRef> clusterRefs) {
+
+        var ref = clusterRefs.stream().filter(cr -> cr.getMetadata().getName().equals(cluster.getSpec().getTargetCluster().getClusterRef().getName())).findFirst();
+
+        if (ref.isEmpty()) {
+            // I should be a condition
```

Comment:
KW: report this with a condition

Reply:
Happy to merge with this deferred to a future PR, as this PR is going to conflict a lot with adding KafkaProxyIngress. Keen to get them both into main quicker so that the bulk of the conflicts are out of the way.